projects/packages/my-jetpack/_inc/utils/get-product-checkout-url.js

```diff
@@ -2,29 +2,20 @@
 /* global myJetpackInitialState */
 
 /**
- * External dependencies
- */
-import { getRedirectUrl } from '@automattic/jetpack-components';
-
-/**
- * Internal dependencies
- */
-import { MY_JETPACK_PRODUCT_CHECKOUT } from '../constants';
-
-/**
- * Return the product redirect URL, according to
- * the Jetpack redirect source, site, path, and redirect_to params.
+ * Return the checkout URL for the given product.
+ * It checkes whether the user is connected to Jetpack.
+ * It doesn't use Jetpack redirect because
+ * it's not possible to pass a valid redirect_to parameter.
+ * In short, and to make it work, it replaces the `=` character with `%3D`.
  *
- * @param {string} product - Checkout product name
+ * @param {string} product          - Checkout product name
+ * @param {boolean} isUserConnected - True when the user is connected Jetpack
  * @returns {string} the redirect URL
  */
-export default function getProductCheckoutUrl( product ) {
-	const { siteSuffix: site, redirectUrl } = window?.myJetpackInitialState || {};
-	const redirect_to = `${ redirectUrl }&product=${ product }`;
-
-	return getRedirectUrl( MY_JETPACK_PRODUCT_CHECKOUT, {
-		site,
-		path: 'jetpack_search',
-		query: `redirect_to=${ redirect_to }`,
-	} );
+export default function getProductCheckoutUrl( product, isUserConnected ) {
+	const { siteSuffix, redirectUrl } = window?.myJetpackInitialState || {};
+	const redirect_to = `${ redirectUrl.replace( /=/g, '%3d' ) }${
```

Comment:
You could use `encodeURIComponent` to handle the `=` to `%3d`, not?

https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent

Reply:
Yes, I think we can use it partially. we can't encode the whole URL. Let me use it at least for the My Jetpack page.

Reply:
We can encode only the `redirectUrl`. This is something that maybe we can address in the Jetpack Redirect system. Addressed [here](https://github.com/Automattic/jetpack/pull/22740/commits/b7c6debf6f3ef4cc2e64a246964909bc526a393c).